rocAL/include/augmentations/audio_augmentations/node_resample.h

```diff
@@ -0,0 +1,43 @@
+/*
+Copyright (c) 2024 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#pragma once
+#include "graph.h"
+#include "node.h"
+#include "rocal_api_types.h"
+
+class ResampleNode : public Node {
+   public:
+    ResampleNode(const std::vector<Tensor *> &inputs, const std::vector<Tensor *> &outputs);
+    ResampleNode() = delete;
+    void init(Tensor *resample_rate, float quality);
+
+   protected:
+    void create_node() override;
+    void update_node() override;
+
+   private:
+    Tensor *_resample_rate;
+    float _quality;
+    vx_array _src_sample_rate_array;
```

Comment:
This can be made local

Reply:
Used in both create_node and update_node